src/main/java/uk/gov/hmcts/reform/logging/exception/AbstractLoggingException.java

```diff
@@ -15,4 +22,28 @@ protected AbstractLoggingException(String message) {
     protected AbstractLoggingException(String message, Throwable cause) {
         super(message, cause);
     }
+
+    private static void triggerBadImplementationLog(Throwable cause) {
+        Throwable invalid = new InvalidExceptionImplementation("AlertLevel is mandatory as per configuration", cause);
+
+        log.error("Bad implementation of '" + cause.getClass().getCanonicalName() + "' in use", invalid);
+    }
+
+    public static AbstractLoggingException getFromLogEvent(ILoggingEvent event) {
+        Throwable eventException = ((ThrowableProxy) event.getThrowableProxy()).getThrowable();
+
+        try {
+            return (AbstractLoggingException) eventException;
```

Comment:
this looks like it casts to an AbstractLoggingException here and then if it fails it does the same thing again

Reply:
if fails, checks the cause. i noticed for spring boot apps it is wrapped in some tomcat container exception (comment below). we preferably want a more informative top level log than that